vm/src/vm/runners/cairo_runner.rs

```diff
@@ -78,24 +78,34 @@ impl From<Vec<MaybeRelocatable>> for CairoArg {
 #[derive(Clone, Default, Debug, PartialEq)]
 pub struct RunResources {
     n_steps: usize,
+    consume_steps: bool,
 }
 
 impl RunResources {
     pub fn new(n_steps: usize) -> Self {
-        RunResources { n_steps }
+        RunResources {
+            n_steps,
+            consume_steps: true,
+        }
     }
 
     pub fn consumed(&self) -> bool {
-        self.n_steps == 0
+        self.consume_steps && self.n_steps == 0
     }
 
     pub fn consume_steps(&mut self) {
-        self.n_steps -= 1;
+        if self.consume_steps {
+            self.n_steps -= 1;
+        }
     }
 
     pub fn get_n_steps(&self) -> usize {
         self.n_steps
     }
+
+    pub fn get_consume_steps(&self) -> bool {
```

Comment:
```suggestion
    pub fn should_consume_steps(&self) -> bool {
```

Reply:
I end up with the n_steps: Option<usize> solution